src/lib/ipfs-utils.js

```diff
@@ -57,54 +57,56 @@ export async function fetchIPFS(uriOrCid) {
-      continue;
-    }
+  try {
+    for await (const file of ipfs.get(cid)) {
+      // If the file type is dir, it's a directory,
+      // so we need inside files
+      if (file.type === 'dir') {
+        continue;
+      }
 
-    if (file.type === 'file') {
-      const content = [];
+      if (file.type === 'file') {
+        const content = [];
 
-      for await (const chunk of file.content) {
-        content.push(chunk);
-      }
+        for await (const chunk of file.content) {
+          content.push(chunk);
+        }
 
-      if (file.path.includes('metadata')) {
-        try {
-          data.metadata = JSON.parse(new TextDecoder().decode(Buffer.concat(content)));
-        } catch (err) {}
-      } else {
-        data.endpoint = defaultIpfsGateway + '/' + file.path;
-
-        const extension = file.path.split('.').pop();
-        // check if the extension exists and is of type `.txt`
-        // to get the text representation by saving bandwith.
-        if (Object.values(FILE_EXTS).includes(extension)) {
+        if (file.path.includes('metadata')) {
           try {
-            data.text = new TextDecoder().decode(Buffer.concat(content));
+            data.metadata = JSON.parse(new TextDecoder().decode(Buffer.concat(content)));
           } catch (err) {}
-        } // if the path name doesn't have .txt extension
-        // or doesn't include path at all, fetch is needed
-        // to determine the type and gets its text if it's text/plain
-        else {
-          const response = await fetch(defaultIpfsGateway + '/' + file.path);
-          if (!response.ok) {
-            data.error = !response.ok;
-          }
+        } else {
+          data.endpoint = defaultIpfsGateway + '/' + file.path;
 
-          const blob = await response.clone().blob();
-
-          if (MIME_TYPES.includes(blob.type)) {
+          const extension = file.path.split('.').pop();
+          // check if the extension exists and is of type `.txt`
+          // to get the text representation by saving bandwith.
+          if (Object.values(FILE_EXTS).includes(extension)) {
             try {
-              data.text = new TextDecoder().decode(content[0]);
+              data.text = new TextDecoder().decode(Buffer.concat(content));
             } catch (err) {}
+          } // if the path name doesn't have .txt extension
+          // or doesn't include path at all, fetch is needed
+          // to determine the type and gets its text if it's text/plain
+          else {
+            const response = await fetch(defaultIpfsGateway + '/' + file.path);
+            if (!response.ok) {
+              data.error = !response.ok;
+            }
+
+            const blob = await response.clone().blob();
+
+            if (MIME_TYPES.includes(blob.type)) {
+              try {
+                data.text = new TextDecoder().decode(content[0]);
+              } catch (err) {}
+            }
           }
         }
       }
     }
-  }
+  } catch(err) {}
```

Comment:
@mathewmeconry given we are try catching, maybe worth to console.error the err so it is easier to debug if something goes wrong

Reply:
good point. for clarification something is off with the dispute 6 which causes ipfs-http-client to fail